core/arch/arm32/kernel/thread.c

```diff
@@ -419,7 +541,41 @@ uint32_t thread_get_id(void)
 	return ct;
 }
 
-void thread_init_handlers(const struct thread_handlers *handlers)
+static void init_handlers(const struct thread_handlers *handlers)
+{
+	thread_std_smc_handler_ptr = handlers->std_smc;
+	thread_fast_smc_handler_ptr = handlers->fast_smc;
+	thread_fiq_handler_ptr = handlers->fiq;
+	thread_svc_handler_ptr = handlers->svc;
+	thread_abort_handler_ptr = handlers->abort;
+	thread_cpu_on_handler_ptr = handlers->cpu_on;
+	thread_cpu_off_handler_ptr = handlers->cpu_off;
+	thread_cpu_suspend_handler_ptr = handlers->cpu_suspend;
+	thread_cpu_resume_handler_ptr = handlers->cpu_resume;
+	thread_system_off_handler_ptr = handlers->system_off;
+	thread_system_reset_handler_ptr = handlers->system_reset;
+}
+
+
+#ifdef CFG_WITH_PAGER
+static void init_thread_stacks(void)
+{
+	/* Do nothing when pager is enabled */
```

Comment:
It seems it would be possible to move `main_init_thread_stacks()` (plat-vexpress/main.c) here?